lib/arm/cpu_features.c

```diff
@@ -167,6 +167,56 @@ static u32 query_arm_cpu_features(void)
 	}
 	return features;
 }
+
+#elif defined(_WIN32)
+
+/*
+ * https://docs.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions
+ * "The ARM64 version of Windows presupposes that it's running on an ARMv8 or
+ * later architecture at all times. Both floating-point and NEON support are
+ * presumed to be present in hardware."
+ * Some (but not all) other features are exposed in various Windows APIs.
+ * However, the cached values of the CP15 regs can also be read out of the
+ * registry.
+ */
+
+#include <Windows.h>
+#include <stdio.h>
+
+/* Note this reads cpu 0 only */
+static bool read_cp_reg(u64* value, u32 reg)
+{
+	DWORD value_len = sizeof(*value);
+	char value_name[12];
+	if (snprintf(value_name, sizeof(value_name), "CP %x", reg) < 0)
+		return false;
+	return RegGetValueA(HKEY_LOCAL_MACHINE,
+		"HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\0", value_name,
+		RRF_RT_REG_QWORD, NULL, value, &value_len) == ERROR_SUCCESS;
+}
+
+static u32 query_arm_cpu_features(void)
+{
+	u32 AA64ISAR0_EL1 = ARM64_SYSREG(3, 0, 0, 6, 0); // ISA Feature Register 0
+	u32 features = ARM_CPU_FEATURE_NEON;
```

Comment:
This all assumes arm64.  I take it that Windows + arm32 is not a thing anymore?  It used to exist, apparently.

Reply:
supposedly, it should be possible to run arm32 on arm64 cpu anyways.